### PR TITLE
Publish to Netlify from CI

### PR DESCRIPTION
**Pull request by Westbrook**
## What I did

1. Update Netlify CLI
2. Build and publish docs from Github Actions 🤞🏼 
3. Save money!

### CODE REVIEW

**Comment by Westbrook** [comment]
Looks like maybe I need a special setting in the build command to purge all the `localhost:8000` references? view-source:https://netlify-publish--open-wc-org.netlify.app/guides/knowledge/styling/index.html

@daKmoR what's Rocket expecting here?

**Comment by bennypowers** [comment]
https://github.com/modernweb-dev/rocket/blob/main/packages/core/src/helpers/index.js

**Comment by bennypowers** [comment]
those vars are provided in the netlify build runner
https://docs.netlify.com/configure-builds/environment-variables/#netlify-configuration-variables

**Comment by bennypowers** [comment]
this might help? https://cli.netlify.com/commands/env

**Comment by Westbrook** [comment]
Is there really no way to provide base URLs to Rocket other than those provided specifically by Netlify? Are consumers not expected to leverage any other tools for publishing their site?

Exposing those env vars manually seemed to help, but some URLs still get published at `localhost:8000`... See line 18: view-source:https://netlify-publish--open-wc-org.netlify.app/guides/knowledge/styling/index.html Thoughts?

**Comment by bennypowers** [comment]
well that function from above is really just a plugin, so the sky's the limit, right?

**Comment by Westbrook** [comment]
@daKmoR any ideas why Rocket process the canonical URL different than other URLs? `<link rel="canonical" href="http://localhost:8000/">` Can track this to a source, even with the `absoluteBaseUrl` corrected in other contexts...

File: .github/workflows/preview.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+      - name: Build the preview URL from the branch name
+        run: echo ::set-output name=url::https://${GITHUB_HEAD_REF}--open-wc-org.netlify.app/
+        id: preview
+
+      - name: Publish to Netlify
+        env:
+          NETLIFY_SITE_ID: ${{ secrets.NETLIFY_SITE_ID }}
+          NETLIFY_AUTH_TOKEN: ${{ secrets.NETLIFY_AUTH_TOKEN }}
+          PREVIEW_URL: ${{ steps.preview.outputs.url }}
```
Comment (daKmoR) [inline .github/workflows/preview.yml:39]: according to [this](https://github.com/modernweb-dev/rocket/blob/main/packages/core/src/helpers/index.js#L13) it's called deploy url? what is preview URL?

and it should also set a context I think

how about this?

```suggestion
          DEPLOY_URL: ${{ steps.preview.outputs.url }}
          CONTEXT: deploy-preview
```

**Review by daKmoR — APPROVED** [review]

**Pull request merged**